### PR TITLE
Enhance the frame for CI

### PR DESCRIPTION
**Pull request by xueli181114**
What I did in the PR:
* Move logger to a package to make it global. Please in your future coding, use logger rather than ```fmt.Println()``` to control the output
* Make Gateway, Region, Version, ChannelGroup to something can be overwritten in profile CI testing to meet future on-demand testing requirement
* Move clusterID initialize to each Describe. Not all cases have a cluster ID because some of them are day1 negative
* Fill clusterID = CI.PrepareRHCSClusterByProfileENV() to make it can ready ENV CLUSTER_ID or get output from output of terraform output
* Move oc command to a new package `openshift` , EXE should only contains the manifest files execution
* Remove garbage code

### CODE REVIEW

**Comment by xueli181114** [comment]
image

**Comment by xueli181114** [comment]
@amalykhi @shellymiron Please review the new updating to the frame. Recommend to create cluster by command 
```
export CLUSTER_PROFILE=rosa-sts-ad; ginkgo run --label-filter day1-prepare tests/e2e
```
So all your day2 cases will use the created cluster which can fit CI

File: tests/utils/constants/config.go [hunk 1]
```diff
@@ -33,10 +31,10 @@ type RHCSconfig struct {
...
 	project := "terraform-provider-rhcs"
-	rootDir := GetEnvWithDefault(CON.WorkSpace, strings.SplitAfter(currentDir, project)[0])
+	rootDir := GetEnvWithDefault(WorkSpace, strings.SplitAfter(currentDir, project)[0])
 
 	// defaulted to staging
-	RHCS.RHCSEnv = GetEnvWithDefault(CON.OCMEnv, RHCS.RHCSEnv)
+	RHCS.RHCSEnv = GetEnvWithDefault(RHCSENV, RHCS.RHCSEnv)
 
 	if os.Getenv("CLUSTER_PROFILE") != "" {
 		RHCS.ClusterProfile = os.Getenv("CLUSTER_PROFILE")
```
Comment (shellymiron) [inline tests/utils/constants/config.go:40]: why not using the function `GetEnvWithDefault`?
it set default value in case of null vars

File: tests/ci/profile_handler.go [hunk 2]
```diff
@@ -97,11 +97,10 @@ func PrepareAccountRoles(token string, accountRolePrefix string, awsRegion strin
 	}
 	args := &EXE.AccountRolesArgs{
 		AccountRolePrefix: accountRolePrefix,
-		// OCMENV            string `json:"ocm_environment,omitempty"`
```
Comment (amalykhi) [inline tests/ci/profile_handler.go:100]: I think we need ocm_environment.
IIUC it is used for applying correct trust policies. Right, @nirarg ?

File: tests/e2e/idps_test.go [hunk 3]
```diff
@@ -5,13 +5,19 @@ import (
...
 	conn "github.com/terraform-redhat/terraform-provider-rhcs/tests/ci"
 	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/cms"
 	con "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
 	exe "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/openshift"
 )
 
 var _ = Describe("TF Test", func() {
+	var clusterID string
+	BeforeEach(func() {
```
Comment (amalykhi) [inline tests/e2e/idps_test.go:18]: Why do we need to get cluster ID before every test case? why not to get in in the "Before suit" - in init_test.go?

Comment (xueli181114) [reply tests/e2e/idps_test.go:18]: Because not all cases need a cluster ID. Especially some day1 negative cases, there is no cluster created at all.

File: tests/ci/profile_handler.go (hunk 2 above)
Comment (xueli181114) [reply tests/ci/profile_handler.go:100]: Really? I remember it was removed? Let me double check.

File: tests/utils/constants/config.go (hunk 1 above)
Comment (xueli181114) [reply tests/utils/constants/config.go:40]: os.Getenv("CLUSTER_PROFILE") will give an empty string if no ENV set. So no need to set a default value.

Comment (xueli181114) [reply tests/utils/constants/config.go:40]: CLUSTER_PROFILE is a must set ENV, we should raise error rather that set a default one.

File: tests/e2e/idps_test.go (hunk 3 above)
Comment (amalykhi) [reply tests/e2e/idps_test.go:18]: But most of the cases will be Day2.  
If I understand the flow - the cluster will be created anyway by profile in the flow, but not every test case will use it. Looks for me overwhelming to set it for every Before Each, instead of - once per test suit

Comment (shellymiron) [reply tests/e2e/idps_test.go:18]: @xueli181114 I believe there won't be any error if there won't be any use for it for some cases,
so it can still be in the `init_test.go` IMO

File: tests/ci/profile_handler.go (hunk 2 above)
Comment (nirarg) [reply tests/ci/profile_handler.go:100]: The `ocm_environment` wasn't removed, there was just added default value `production` in the module, so user doesn't have to indicate this.
We added new data source `rhcs_info` that will replace this requirement, its not planed to be changed in this module, the new WIP module is going to include this

Comment (xueli181114) [reply tests/ci/profile_handler.go:100]: Yes,  I checked the code. It is indeed required. Will fix it.

File: tests/e2e/idps_test.go (hunk 3 above)
Comment (xueli181114) [reply tests/e2e/idps_test.go:18]: OK. I checked terraform output won't return error when there is no resources creation. Moving it back to init_test.go

File: tests/ci/profile_handler.go (hunk 2 above)
Comment (xueli181114) [reply tests/ci/profile_handler.go:100]: Fixed this part. I can create a cluster on production now with env set.
To make all resources using same env, I set it in exec package, rather than profile handler.

File: tests/e2e/idps_test.go (hunk 3 above)
Comment (xueli181114) [reply tests/e2e/idps_test.go]: Updated. It had been moved back to init_test.go

File: tests/README.md [hunk 4]
```diff
@@ -121,9 +121,34 @@ This allows run by case filter to simulate CI. Anybody can customize the case la
   * Export the token
     * `export RHCS_TOKEN=<rhcs token>` #If it had been run, then you can skip
   * Export the label filter to locate the preparation case
-    * `export LabelFilter="destroy"` #This filter only choose all critical and high priority cases with day1-post and day2
+    * `export LabelFilter="destroy"` #This filter only choose the destroy case for resources destroy
   * Run make command
     * `make e2e_test`
+### To run on local simulate CI without make file
+This allows run by case filter to simulate CI. Anybody can customize the case label filter to filter out the cases want to run
```
Comment (bardielle) [inline tests/README.md:128]: I didn't understand it. 
@EricPonvelle can you rephrase it or help @xueli181114 with it?

Comment (bardielle) [reply tests/README.md:128]: Maybe something like that: 
```suggestion
### Running a local CI simulation
This feature allows for running tests through a case filter to simulate CI. Anyone can customize the case label filter to select the specific cases that would be run. 
```

File: tests/README.md [hunk 5]
```diff
@@ -121,9 +121,34 @@ This allows run by case filter to simulate CI. Anybody can customize the case la
   * Export the token
     * `export RHCS_TOKEN=<rhcs token>` #If it had been run, then you can skip
   * Export the label filter to locate the preparation case
-    * `export LabelFilter="destroy"` #This filter only choose all critical and high priority cases with day1-post and day2
+    * `export LabelFilter="destroy"` #This filter only choose the destroy case for resources destroy
   * Run make command
     * `make e2e_test`
+### To run on local simulate CI without make file
```
Comment (amalykhi) [inline tests/README.md:127]: Are the flows with  `make e2e_test` command still relevant?

Also it fails on • [PANICKED] [50.411 seconds]
TF Test Create cluster test [It] CreateClusterByProfile [day1-prepare]
/Users/amalykhi/repos/terraform-provider-rhcs/tests/e2e/cluster_creation.go:13

  [PANICKED] Test Panicked
  In [It] at: /opt/homebrew/Cellar/go/1.20.4/libexec/src/runtime/iface.go:262 @ 10/18/23 15:36:56.799

  interface conversion: interface {} is nil, not []interface {}

Comment (xueli181114) [reply tests/README.md:127]: Good to know you fixed it. Resolve it.

File: tests/utils/log/logger.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package log
+
+import (
+	logging "github.com/sirupsen/logrus"
+)
+
+func GetLogger() *logging.Logger {
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/utils/log/logger.go:7]: It's future's work, not high priority.

File: tests/ci/profile_handler.go [hunk 7]
```diff
@@ -215,25 +213,35 @@ func GenerateClusterCreationArgsByProfile(token string, profile *Profile) (clust
...
-		if vpcOutput.ClusterPrivateSubnets == nil {
-			err = fmt.Errorf("error when creating the vpc, check the previous log. The created resources had been destroyed")
-			return
-		}
-		if profile.PrivateLink {
-			clusterArgs.AWSSubnetIDs = vpcOutput.ClusterPrivateSubnets
+		// Supports ENV set passed to make cluster provision more flexy in prow
+		// Export the subnetIDs via env variable if you have existing ones export SubnetIDs=<subnet1>,<subnet2>,<subnet3>
+		// Export the availability zones via env variable export AvailabilitiZones=<az1>,<az2>,<az3>
+		if os.Getenv("SubnetIDs") != "" && os.Getenv("AvailabilitiZones") != "" {
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/ci/profile_handler.go:219]: @yasun1 No, there is one already to overwrite the profile by ENV those are highlevel. This part is only exposed for shared vpc in case anybody wants to pass the value from other steps.

File: tests/ci/profile_handler.go [hunk 8]
```diff
@@ -13,6 +13,7 @@ import (
 	CON "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
 	EXE "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
 	HELPER "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/helper"
+	. "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/log"
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/ci/profile_handler.go:16]: I prefer not to have so many log.Logger.xxx(xxx). There won't be too many functions defined in the package.
